### PR TITLE
Add additional path expression tests

### PR DESCRIPTION
**Pull request by am357**
*Issue #, if available:* #121

*Description of changes:*

Add additional path expression tests to the conformance tests.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jpschorr** [comment]
These examples seem to be straight from [section 4](https://partiql.org/assets/PartiQL-Specification.pdf#section.4) of the spec, right?

**Review by jpschorr — APPROVED** [review]

**Comment by am357** [comment]
> These examples seem to be straight from [section 4](https://partiql.org/assets/PartiQL-Specification.pdf#section.4) of the spec, right?

Yes, some of there are. I'm going to add a new comment with some more.

File: partiql-test-data/pass/parser/primitives/path-expression.ion [hunk 1]
```diff
@@ -137,3 +137,63 @@ parse::{
...
+    assert: [
+        {
+            result: ParseOk
+        },
+    ]
+}
+
+parse::{
+    name: "path expression on select query",
+    statement: "(SELECT a from en).b",
```
Comment (alancai98) [inline partiql-test-data/pass/parser/primitives/path-expression.ion:173]: nit: for consistency, could capitalize the keyword `FROM`

```suggestion
    statement: "(SELECT a FROM en).b",
```

**Review by alancai98 — APPROVED** [review]
Left a very minor nit. Looks good otherwise.

File: partiql-test-data/fail/parser/select/select-from-where.ion [hunk 2]
```diff
@@ -0,0 +1,7 @@
+parse::{
```
Comment (am357) [inline partiql-test-data/fail/parser/select/select-from-where.ion:1]: This appears to have come from a rebase.

**Review by alancai98 — APPROVED** [review]

**Review by jpschorr — APPROVED** [review]

**Pull request merged**